src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,113 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final double GROW_COEFFICIENT = 0.5;
+    private static final byte DEFAULT_SIZE = 10;
+    private int size = DEFAULT_SIZE;
+    private int currentIndex;
+    private T[] storage;
+
+    @SuppressWarnings("unchecked")
+    public ArrayList() {
+        storage = (T[]) new Object[DEFAULT_SIZE];
+        currentIndex = 0;
+    }
+
     @Override
     public void add(T value) {
-
+        shouldGrow();
```

Comment:
```suggestion
        resizeArray();
```
Let’s  give a more appropriate  name for this method